packages/integration-tests/src/e2e-flow/e2e_flow.test.ts

```diff
@@ -0,0 +1,440 @@
+          [
+            'tsc',
+            '--noEmit',
+            '--skipLibCheck',
+            // pointing the project arg to the amplify backend directory will use the tsconfig present in that directory
+            '--project',
+            amplifyPathPrefix,
+          ],
+          {
+            cwd: tempDir,
+            stdio: 'inherit',
+          }
+        );
+
+        if (PACKAGE_MANAGER_EXECUTABLE.startsWith('yarn')) {
+          await execa('yarn', ['add', 'aws-cdk', 'aws-cdk-lib', 'constructs'], {
+            cwd: tempDir,
+            stdio: 'inherit',
+          });
+          if (PACKAGE_MANAGER_EXECUTABLE === 'yarn-modern') {
+            await execa(
+              'yarn',
+              [
+                'add',
+                '-D',
+                'tsx',
+                'graphql',
+                'pluralize',
+                'zod',
+                '@aws-amplify/platform-core',
+              ],
+              {
+                cwd: tempDir,
+                stdio: 'inherit',
+              }
+            );
+
+            await execa('node', ['--version'], {
+              cwd: tempDir,
+            });
+          }
+        }
+
+        // assert that project synthesizes successfully
+        await execa(
+          packageManagerExecutable === 'npm' ? 'npx' : packageManagerExecutable,
+          [
+            'cdk',
+            'synth',
+            '--context',
+            `amplify-backend-namespace=123`,
+            '--context',
+            `amplify-backend-name=sandboxName`,
+            '--context',
+            `amplify-backend-type=sandbox`,
+            '--app',
+            "'npx tsx amplify/backend.ts'",
+            '--quiet',
+          ],
+          {
+            cwd: tempDir,
+            stdio: 'inherit',
+          }
+        );
+
+        await execa(
+          packageManagerExecutable === 'npm' ? 'npx' : packageManagerExecutable,
+          [
+            'amplify',
+            'pipeline-deploy',
+            '--branch',
+            branchBackendIdentifier.name,
+            '--appId',
+            branchBackendIdentifier.namespace,
+          ],
+          { cwd: tempDir, stdio: 'inherit', env: { CI: 'true' } }
+        );
+
+        const clientConfigStats = await fsp.stat(
+          await getClientConfigPath(tempDir)
+        );
+
+        assert.ok(clientConfigStats.isFile());
+
+        const testBranchDetails = await amplifyAppPool.fetchTestBranchDetails(
+          testBranch
+        );
+        assert.ok(
+          testBranchDetails.backend?.stackArn,
+          'branch should have stack associated'
+        );
+        assert.ok(
+          testBranchDetails.backend?.stackArn?.includes(
+            branchBackendIdentifier.namespace
+          )
+        );
+        assert.ok(
+          testBranchDetails.backend?.stackArn?.includes(
+            branchBackendIdentifier.name
+          )
+        );
+      });
+    });
+  });
+
+  void describe('fails fast', () => {
```

Comment:
we can remove this block too if we're testing in separate file.

Reply:
Removed